### PR TITLE
Improve `engine.reflection`

### PR DESCRIPTION
**Pull request by bryanforbes**
### Description
Improve `engine.reflection`

### Checklist

This pull request is:

- [ ] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [x] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

File: sqlalchemy-stubs/engine/reflection.pyi [hunk 1]
```diff
@@ -1,69 +1,83 @@
...
-        self, sequence_name: Any, schema: Optional[Any] = ...
-    ): ...
-    def get_sorted_table_and_fkc_names(self, schema: Optional[Any] = ...): ...
-    def get_temp_table_names(self): ...
-    def get_temp_view_names(self): ...
+        self, sequence_name: str, schema: Optional[str] = ...
+    ) -> bool: ...
+    def get_sorted_table_and_fkc_names(
+        self, schema: Optional[str] = ...
+    ) -> List[Tuple[Optional[str], List[Tuple[str, str]]]]: ...
```
Comment (CaselIT) [inline sqlalchemy-stubs/engine/reflection.pyi:36]: python generic code syncacs is really ugly compared to other languages like ts :(

**Review by CaselIT — APPROVED** [review]

File: sqlalchemy-stubs/engine/reflection.pyi (hunk 1 above)
Comment (bryanforbes) [reply sqlalchemy-stubs/engine/reflection.pyi:36]: It can be, but even in TS you can do `Array<Array<string | null, Array<Array<string, string>>>>`

Comment (CaselIT) [reply sqlalchemy-stubs/engine/reflection.pyi:36]: sure, hopefully 3.9+ is nice since you can use lowercase stuff and `|` for unions

Comment (bryanforbes) [reply sqlalchemy-stubs/engine/reflection.pyi:36]: Using `|` for unions and `Optional` is such a huge improvement

**Pull request merged**